File: assignments/a1/q2/Java/Scanin.java
//Scanin:top//
//Scanin:import//
import java.util.*;

// <scanin> ::= <type> TITLE <rule>
public class Scanin extends _Start /*Scanin:class*/ {

    public static final String $className = "Scanin";
    public static final String $ruleString =
        "<scanin> ::= <type> TITLE <rule>";

    public Type type;
    public Rule rule;

    public Scanin(Type type, Rule rule) {
//Scanin:init//
        this.type = type;
        this.rule = rule;
    }

    public static Scanin parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<scanin>", scn$.lno);
        Type type = Type.parse(scn$, trace$);
        scn$.match(Token.Match.TITLE, trace$);
        Rule rule = Rule.parse(scn$, trace$);
        return new Scanin(type, rule);
    }

//Scanin//
}

File: assignments/a1/q2/Python/Scanin.py
#Scanin:top#
#Scanin:import#

# <scanin> ::= <type> TITLE <rule>
class Scanin(_Start): #Scanin:class#

    className = "Scanin"
    ruleString = "<scanin> ::= <type> TITLE <rule>"
    type = None
    rule = None

    def __init__(type, rule):
        #Scanin:init#
        self.type = type
        self.rule = rule

#Scanin#
